SKYKitChat/Classes/UI/ConversationView/SKYChatConversationViewController.swift

```diff
@@ -342,6 +370,46 @@ extension SKYChatConversationViewController {
 
         self.skygear.chatExtension?.subscribeToMessages(in: self.conversation!,
                                                         handler: self.messageChangeHandler!)
+
+    }
+
+    open func subscribeTypingIndicatorChanges() {
+        if self.typingIndicatorChangeHandler == nil {
+            self.typingIndicatorChangeHandler = {(indicator) in
+                // invalidate the existing timer
+                if let timer = self.typingIndicatorPromptTimer {
+                    timer.invalidate()
+                    self.typingIndicatorPromptTimer = nil
+                }
+
+                switch indicator.userIDs.count {
+                case 0:
+                    self.hideTypingIndicator()
+                case 1:
+                    // Show indicator if not I am typing
+                    if let typingID = SKYRecordID(canonicalString: indicator.userIDs.first!) {
+                        if typingID.recordName != self.senderId {
+                            self.displayTypingIndicator()
+                        }
+                    }
+                    break
+                default:
+                    self.displayTypingIndicator()
+                }
```

Comment:
This switch statement took me a minute to understand...
I am not familiar with the indicator response, but I guess there is no guarantee that my user id always comes first?

Wouldn't it be simplier if you do this?
```
let shouldShow = indicator.userIDs
    .flatMap { SKYRecordID(canonicalString: $0) }
    .filter { $0 != self.senderId }
    .count > 0
```